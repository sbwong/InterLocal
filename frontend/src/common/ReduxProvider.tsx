import React, { ReactNode } from "react";

import { Provider } from "react-redux";
import { store } from "../app/store";

export interface ReduxProviderProps {
	children: ReactNode;
}

export default function ReduxProvider(props: ReduxProviderProps) {
	return <Provider store={store}>{props.children}</Provider>;
}
