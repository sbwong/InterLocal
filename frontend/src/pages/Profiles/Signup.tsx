import Box from "@material-ui/core/Box";
import React from 'react';
import SignupBox from './components/SignupBox';

export function Signup() {

    return (
        <Box paddingTop="30px" marginTop="50px" marginBottom="100px" justifyContent="center">
            <SignupBox />
        </Box>
    );
}
