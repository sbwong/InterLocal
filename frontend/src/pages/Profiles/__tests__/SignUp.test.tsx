import React, { ReactNode } from "react";
import ReduxProvider, {
	ReduxProviderProps
} from "../../../common/ReduxProvider";

import ReactDOM from "react-dom";
import { Signup } from "../Signup";
import { render } from "@testing-library/react";

it("renders without crashing", () => {
	const div = document.createElement("div");

	ReactDOM.render(
		<ReduxProvider>
			<Signup />
		</ReduxProvider>,
		div
	);
});

it("renders signup page", () => {
	const { signup } = render(
		<ReduxProvider>
			<Signup />
		</ReduxProvider>
	);
	expect(signup).toMatchSnapshot();
});
