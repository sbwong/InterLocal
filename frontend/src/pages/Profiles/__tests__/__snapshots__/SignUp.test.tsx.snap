// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders signup page 1`] = `undefined`;
